engine/deeplearning/src/main/java/org/opennms/alec/engine/deeplearning/ClasspathUtils.java

```diff
@@ -97,19 +105,56 @@ static void copyResourcesRecursively(URL originUrl, File destination) throws IOE
         }
     }
 
-    private static void copyJarResourcesRecursively(File destination, JarURLConnection jarConnection ) throws IOException {
+    private static void copyJarResourcesRecursively(File destination, JarURLConnection jarConnection) throws IOException {
+        int totalSizeArchive = 0;
+        int totalEntryArchive = 0;
+
         final JarFile jarFile = jarConnection.getJarFile();
-        for (JarEntry entry : Collections.list(jarFile.entries())) {
-            if (entry.getName().startsWith(jarConnection.getEntryName())) {
-                final String fileName = removeStart(entry.getName(), jarConnection.getEntryName());
-                if (!entry.isDirectory()) {
-                    try (InputStream entryInputStream = jarFile.getInputStream(entry)) {
-                        FileUtils.copyInputStreamToFile(entryInputStream, new File(destination, fileName));
+        Enumeration<JarEntry> entries = jarFile.entries();
+        while (entries.hasMoreElements()) {
+            ZipEntry ze = entries.nextElement();
+            try (InputStream in = new BufferedInputStream(jarFile.getInputStream(ze))) {
+                totalEntryArchive++;
+
+                int nBytes = -1;
+                byte[] buffer = new byte[2048];
+                double totalSizeEntry = 0;
+
+                while ((nBytes = in.read(buffer)) > 0) {
+                    if (ze.getName().startsWith(jarConnection.getEntryName())) {
+                        final String fileName = removeStart(ze.getName(), jarConnection.getEntryName());
```

Comment:
It looks like this could be done before the while loop. Also, if all of the zip entries are retrieved from the jarConnection, is this if statement needed?

Reply:
you're right, fixed